app/js/main.js

```diff
@@ -328,17 +333,55 @@ function startup() {
 
   // Preview a captured frame
   frameReelRow.addEventListener("click", function(e) {
-    if (e.target.className === "frame-reel-img") {
-      if (winMode !== "playback") {
-        switchMode("playback");
+    if (e.target.className.includes("frame-reel-img")) {
+      // For an unselected, non-keyframe
+      if (!e.target.className.includes("selected")) {
+        // Remove previous selection
+        _removeFrameReelSelection();
+
+        // Highlight the clicked image
+        e.target.classList.add("selected");
+        if (winMode !== "playback") {
+          switchMode("playback");
+        }
+
+        // Display the selected frame
+        var imageID = parseInt(e.target.id.match(/^img-(\d+)$/)[1], 10);
+        _displayFrame(imageID);
+
+      } else if (e.target.className === "frame-reel-img selected") {
+        addKeyframe("start", curSelectedFrame);
+
+      } else if (e.target.className.includes("keyframe")) {
+        // If the current start keyframe is double clicked it is removed
+        if (parseInt(e.target.id.slice(4), 10) === curStartKeyframe) {
+          removeKeyframe("start");
+        }
       }
+    }
+  });
+
+  // Add a start keyframe
+  btnStartKeyframe.addEventListener("click", function(e) {
+    if (btnStartKeyframe.hasAttribute("disabled")) {
+      return false;
+    } else if (curSelectedFrame === curStartKeyframe) {
+      removeKeyframe("start");
+    } else {
+      addKeyframe("start", curSelectedFrame);
+    }
+  });
 
-      // Display the selected frame
-      var imageID = parseInt(e.target.id.match(/^img-(\d+)$/)[1], 10);
-      _displayFrame(imageID);
+  // Remove the start keyframe
+  btnClearKeyframe.addEventListener("click", function(e) {
```

Comment:
Code structure tip: If you have a condition that `return`s to stop the rest of the function from running, you can eliminate the `else` clause.

```js
btnClearKeyframe.addEventListener("click", function(e) {
    if (!curStartKeyframe) {
      return false;
    }
    removeKeyframe("start");
  });
```